Tools/PIDML/pidMLProducer.cxx

```diff
@@ -190,11 +190,39 @@ struct PidMlProducer {
     }
   }
 
+  template <typename T>
+  float getTOFSignal(T const& track)
+  {
+    return tofMissing(track) ? std::numeric_limits<float>::quiet_NaN() : track.tofSignal();
+  }
+
+  template <typename T>
+  float getTOFBeta(T const& track)
+  {
+    return tofMissing(track) ? std::numeric_limits<float>::quiet_NaN() : track.beta();
+  }
+
+  template <typename T>
+  float getTRDSignal(T const& track)
+  {
+    return trdMissing(track) ? std::numeric_limits<float>::quiet_NaN() : track.trdSignal();
+  }
+
+  template <typename T>
+  uint8_t getTRDPattern(T const& track)
+  {
+    return trdMissing(track) ? static_cast<uint8_t>(0U) : track.trdPattern();
+  }
+
   void processDataML(MyCollisionML const& /*collision*/, BigTracksDataML const& tracks)
   {
     for (const auto& track : tracks) {
-      pidTracksTableDataML(track.tpcSignal(), track.trdSignal(), track.trdPattern(),
-                           track.tofSignal(), track.beta(),
+      if (!track.hasTPC()) {
```

Comment:
jak mamy filtrowanie po global tracks, to nie potrzebujemy sprawdzać dodatkowo hasTPC